flagship/src/app/campaign/campaign-locations/campaign-locations.component.ts

```diff
@@ -60,25 +60,18 @@ export class CampaignLocationsComponent implements OnInit, OnChanges {
 
 
 
-  public getObjectiveNames(objectives: number[]): string {
+  public getObjectiveNames(location: CampaignLocation, objectives: number[]): string {
     if (!objectives) return 'None';
 
     return this
       .objectiveFactory
       .getObjectivesByIds(objectives)
       .map(objective => {
-        return objective.name + (this.isLocationCampaignObjectUsed(objective) ? ' [Already Played]' : '')
+        return objective.name + (location.playedCampaignObjectives.includes(objective.id) ? ' [Already Played]' : '')
       })
       .join(', ');
   }
 
-  private isLocationCampaignObjectUsed(objective: Objective): boolean {
```

Comment:
I inlined this function, as after reducing it to only for the given location, it felt small enough to warrant being inlined.